File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        hash = {}

        for i in range(0, len(nums)):
            diff = target - nums[i]
            if diff in hash:
                return [hash[diff], i]
            hash[nums[i]] = i

        return [-1.-1]
        

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        for(int i=0; i<nums.length/2; i++){
            for(int j=i+1; j<nums.length; j++){
                if(nums[i]+nums[j]==target){
                    int[] res = {i, j};
                    return res;
                }
                if(nums[nums.length-i-1] + nums[nums.length-j-1] == target){
                    return new int[]{nums.length-j-1, nums.length-i-1};
                }  
            }
        }

        return null;
    }
}